### PR TITLE
Style: Suppress new() fixer in main library projects

### PR DESCRIPTION
**Pull request by Wraith2**
Suppresses the fixer for `A a = new A()` to `A a = new()` in the main library projects but allows it in the tools and testing projects. 

My preference is for the more formal style of specifying the type name for the constructor calls. It makes no difference to the functionality I just find it easier to read

### CODE REVIEW

**Comment by johnnypham** [comment]
Did you also want to suppress the rule for scoped using statements?

**Review by johnnypham — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

**Review by DavoudEshtehari — APPROVED** [review]

**Comment by DavoudEshtehari** [comment]
Could you apply these settings on `FunctionalTests` and `ManualTests` projects too?

**Comment by Wraith2** [comment]
I can but I didn't feel that it was a problem to have the terser syntax in the tests. If we're going to disallow it everywhere then the editor config changes can just go into the root level.

**Comment by DavoudEshtehari** [comment]
In my understanding, they're preferences with suggestion severity which means we can use the new styles. If I'm correct I'd rather have the same rule in all projects.
Since I believe we're on the same page about these code suggestions, I leave it to you. 😉

**Comment by JRahnama** [comment]
> Could you apply these settings on `FunctionalTests` and `ManualTests` projects too?

I am not a fan of this suggestion. We need to be more open in our test lab. Like `using block` in the main library is good, but we need to be able to practice new features/styles in our test lab. It is related to learning and testing.

File: src/Microsoft.Data.SqlClient/netfx/src/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# editorconfig.org
+
+# top-most EditorConfig file
+root = false
+
+[*.cs]
+
+# IDE0090: Use 'new(...)'
+csharp_style_implicit_object_creation_when_type_is_apparent = false
```
[the remark this answers is not in the document]
Comment (Wraith2) [reply src/Microsoft.Data.SqlClient/netfx/src/.editorconfig]: Done.

**Comment by DavoudEshtehari** [comment]
> > Could you apply these settings on `FunctionalTests` and `ManualTests` projects too?
> 
> I am not a fan of this suggestion. We need to be more open in our test lab. Like `using block` in the main library is good, but we need to be able to practice new features/styles in our test lab. It is related to learning and testing.

**Review by DavoudEshtehari — APPROVED** [review]

**Pull request merged**